src/auth/AuthServiceHandler.cc

```diff
@@ -26,6 +27,8 @@ AuthServiceHandler *get_auth_service_handler(int type, CephContext *cct, KeyServ
     return new CephxServiceHandler(cct, ks);
   case CEPH_AUTH_NONE:
     return new AuthNoneServiceHandler(cct);
+  case CEPH_AUTH_UNKNOWN:
+    return new AuthUnknownServiceHandler(cct);
```

Comment:
please take a look at https://github.com/ceph/ceph/blob/master/src/mon/AuthMonitor.cc#L423, if none of the auth is`supported`, `CEPH_AUTH_UNKNOWN` is returned by `AuthMethodList::pick()`. with your change, `get_auth_service_handler()` will return an instance of `AuthUnknownServiceHandler`, which is apparently not `nullptr`. so `start = true`, and the code proceeds.  this just breaks the existing logic, which will return `-ENOTSUP` in this case.